src/pytest_cov/engine.py

```diff
@@ -147,8 +147,11 @@ def finish(self):
 
         self.unset_env()
         self.cov.stop()
+        self.cov.save()
+        self.cov.load()
         self.cov.combine()
         self.cov.save()
+
```

Comment:
I think the issue that's causing the duplicates is that you're not setting `self.cov = self.combining_cov` before doing the `load()`.